flink-connector-aws/flink-connector-aws-kinesis-streams/src/main/java/org/apache/flink/connector/kinesis/sink/KinesisStreamsSinkWriter.java

```diff
@@ -167,8 +204,18 @@ class KinesisStreamsSinkWriter<InputT> extends AsyncSinkWriter<InputT, PutRecord
         this.streamArn = streamArn;
         this.metrics = context.metricGroup();
         this.numRecordsOutErrorsCounter = metrics.getNumRecordsOutErrorsCounter();
-        this.httpClient = AWSGeneralUtil.createAsyncHttpClient(kinesisClientProperties);
-        this.kinesisClient = buildClient(kinesisClientProperties, this.httpClient);
+
+        this.kinesisClientProvider = kinesisClientProvider;
+
+        if (kinesisClientProvider != null) {
+            // Use the provided client for testing
+            this.httpClient = null;
+            this.kinesisClient = kinesisClientProvider.get();
+        } else {
+            // Create a new client as before
+            this.httpClient = AWSGeneralUtil.createAsyncHttpClient(kinesisClientProperties);
+            this.kinesisClient = buildClient(kinesisClientProperties, this.httpClient);
+        }
```

Comment:
Nit: These can be pushed to their own constructors so you don't have to pass in `null` or do a `null` check.